### PR TITLE
 NEW possibility to defined rounding rules by currency

### PR DESCRIPTION
**Pull request by hregis**
@eldy 
https://github.com/Dolibarr/dolibarr/issues/9816  work in progress

### CODE REVIEW

File: htdocs/admin/limits.php [hunk 1]
```diff
@@ -66,29 +72,53 @@
...
+	{
+		while ($obj = $db->fetch_object($resql))
+		{
+			$aCurrencies[] = $obj->code;
+		}
+
+		if (! empty($aCurrencies) && count($aCurrencies) > 1)
+		{
+			$head = multicurrencyLimitPrepareHead($aCurrencies);
+			dol_fiche_head($head, $currencycode, '', -1, "multicurrency");
```
Comment (eldy) [inline htdocs/admin/limits.php]: You introduce here a dol_fiche_head, you should probably introduce a dol_fiche_end too on same condition, shouldn't it ?

Comment (hregis) [reply htdocs/admin/limits.php]: @eldy ok, but did you receive my question about these changes?

Comment (hregis) [reply htdocs/admin/limits.php]: @eldy 
the trouble is that the functions that integrate these constants, do not
do not get back, the code of the currency!
do you see a solution without modifying all the "prices",
"price2num", "calcul_price_total" and other codes using these constants?

**Comment by eldy** [comment]
Note that in v11, some changes were done in file price.lib.php, method calcul_price_total to manage this.
However, you must provide the value of the currency code ('EUR', 'USD', ...) to the function calcul_price_total (currently this information is not provided)

**Pull request merged**